### PR TITLE
update milestone - internal review

### PR DESCRIPTION
**Pull request by yz89**
Hello, W3F Committee,

After several weeks of development, we find that the progress of the project deviates a little from the milestone, so we need to update the milestone content.

Grant Application PR: https://github.com/w3f/Open-Grants-Program/pull/329

The main plan of milestone1-1.a is to receive cross-chain assets through XCM and integrate them into Lending Market. Such as moving KSM/ DOT from RelayChain to Parachain and USDT from Statemint to Parchain. But so far neither Rococo-v1 nor the StateMint branch has been able to set up a stable testing environment for our plans.

Regarding the XCM. We have done a considerable amount of work. Please check the links below:
- PR: https://github.com/parallel-finance/parallel/pull/88
- Article: https://www.notion.so/parallelfinance/XCM-91168ae10ab8415996737c52138ef5b4

Over the past few weeks, we've developed some new features, such as
- Redesigned the interest rate mode
- Added the Liquidate Pallet
- Added Price Pallet

We wanted to adjust the Milestone to be able to deliver Milestone 1 now.

### CODE REVIEW

File: applications/Parallel.md [hunk 1]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
--   **Costs:** 5000 USD
+-   **FTE:** 1.8
+-   **Costs:** 4000 USD
 
 The major deliverable of for this milestone:
 
--   Develop cross-chain capabilities between DOT and KSM tokens
--   Users will be able to deposit DOT tokens as collateral and borrow KSM tokens through our web app
+-   Implement a cross-asset lending protocol.
+-   Support automatic liquidation by using off-chain worker.
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Support automatic liquidation by using an off-chain worker.

File: applications/Parallel.md [hunk 2]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
-|     4. | User Testing               | We will conduct user testing to improve our product's UX and UI, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users use the v1 platform and provide a summary of the findings and improvements made based on insights.                                                                            |
+|    1.a | Substrate module: Loans pallet | Loans pallet will be implemented as a cross-asset lending protocol which offers lending and borrowing by using floating rate model.                                                                                     |
+|    1.b | Substrate module: Liquidation pallet | We will implement a liquidation solution build with Substrate Offchain Worker, which will calculates the health factor of each borrower's account and send a liquidation transaction automatically.                                                                                     |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: We will implement a liquidation solution build with Substrate Off-chain Worker, which will calculate the health factor of each borrower's account and send a liquidation transaction automatically.

File: applications/Parallel.md [hunk 3]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    1.c | Substrate module: Price aggregation oracle   |  We will need to create a on-chain price aggregation pallet for the oracle prices from multiple source that is queried by off-chain workers.                                                                            |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: We will need to create an on-chain price aggregation pallet for the oracle prices from multiple sources that is queried by off-chain workers.

File: applications/Parallel.md [hunk 4]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    2.a | Integration with front-end | We will integrate our existing front end to the finalized substrate backend                                                                                                                                                   |
+|    2.b | Article/Tutorial           | We will create an article and a video demo that explains how users can start using the platform in lending and borrowing for DOT or KSM                                                                                      |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: We will create an article and a demo video that will explain how users can start using the platform for lending and borrowing DOT or KSM tokens.

File: applications/Parallel.md [hunk 5]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    1.a | Substrate module: Loans pallet | Loans pallet will be implemented as a cross-asset lending protocol which offers lending and borrowing by using floating rate model.                                                                                     |
+|    1.b | Substrate module: Liquidation pallet | We will implement a liquidation solution build with Substrate Offchain Worker, which will calculates the health factor of each borrower's account and send a liquidation transaction automatically.                                                                                     |
+|    1.c | Substrate module: Price aggregation oracle   |  We will need to create a on-chain price aggregation pallet for the oracle prices from multiple source that is queried by off-chain workers.                                                                            |
+|    2.a | Integration with front-end | We will integrate our existing front end to the finalized substrate backend                                                                                                                                                   |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add `.` at the end of line.

File: applications/Parallel.md [hunk 6]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    2.b | Article/Tutorial           | We will create an article and a video demo that explains how users can start using the platform in lending and borrowing for DOT or KSM                                                                                      |
+|    3. | Docker                     | We will provide a dockerfile to demonstrate the lending functionality                                                                                                                                                |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add `.` at the end of the line.

File: applications/Parallel.md [hunk 7]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    4. | Testing and Documentation  | While we develop our module, we will ensure that our code has significant test coverage (>80%) to ensure quality and provide explanations on how the functions work for the community to review, as well as craft test guides |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add . at the end of line.

File: applications/Parallel.md [hunk 8]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
+|    5. | User Testing               | We will conduct user testing to improve our product's UX and UI, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users use the v1 platform and provide a summary of the findings and improvements made based on insights.                                                                            |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: We will conduct user testing to improve our product's UI and UX, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users who will use the v1 platform and provide a summary of the findings and improvements made based on insights.

File: applications/Parallel.md (hunk 3 above)
Comment (ayushmishra2005) [inline applications/Parallel.md]: We should also tell about the staking module.

File: applications/Parallel.md [hunk 9]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
+|    1.a | Substrate module: Staking DOT       | We will implement the on chain staking pool for the staking process from the nominators.                                                                                                                         |
+|    1.b | Substrate module: Unstaking DOT     | We will implement a 28 days locking period for the unstaking process of DOT tokens from validators.                                                                                                                                                                                                  |
+|    1.c | Substrate module: Exchange Rate     | We will implement the process for issuing and trading DOT for xDOT with the users based on our exchange rate                                                                                                                                                                                      |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add `.` at the end of line.

File: applications/Parallel.md [hunk 10]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
 |    1.d | Substrate module: Slashing Scenario | Staking tokens does have inherent risks of being slashed. We will implement our model that will change the exchange rate in case of slashing scenarios                                                                                                                                              |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add `.` at the end of line.

File: applications/Parallel.md [hunk 11]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
-|    1.b | Substrate module: Unstaking DOT     | We will implement a 28 days locking period for the unstaking process of DOT tokens from validators                                                                                                                                                                                                  |
-|    1.c | Substrate module: Exchange Rate     | We will implement a the process for issuing and trading DOT for xDOT with the users based on our exchange rate                                                                                                                                                                                      |
+|    1.a | Substrate module: Staking DOT       | We will implement the on chain staking pool for the staking process from the nominators.                                                                                                                         |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: on chain => on-chain

File: applications/Parallel.md [hunk 12]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
-|     4. | User Testing                        | We will conduct user testing to improve our product's UX and UI, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users use the v1 platform and provide a summary of the findings and improvements made based on insights                                                                                                                |
+|    2.a | Validator evaluation schema         | We will design the validator evaluation schema to select the outstanding validators.                                                                                                                         |
+|    2.b | Offchain client                     | We will set up multi-signature account on relaychain, and operate the `bond` and `unbond` dispatchables by using offchain client.                                                |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: offchain => off-chain

File: applications/Parallel.md [hunk 13]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
+|    3. | Integration with front-end          | We will integrate our existing front end to the finalized substrate backend                                                                                                                                                                                                                         |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add . at the end of line.

File: applications/Parallel.md [hunk 14]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
+|    4. | Article/Tutorial                    | We will create an article and a video demo that explains how users can start using the platform for staking for DOT and KSM. Additionally, we will also create tutorials to walk show the users how they can earn "double interests" through staking and lending at the same time via xDOT and xKSM |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: We will create an article and a demo video that will explain how users can start using the platform for staking DOT and KSM. Additionally, we will also create tutorials for users to explain how they can earn "double interests" through staking and lending at the same time via xDOT and xKSM.

File: applications/Parallel.md [hunk 15]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
+|    5. | Testing and Documentation           | While we develop our module, we will ensure that our code has significant test coverage (>80%) to ensure quality and provide explanations on how the functions work for the community to review, as well as craft test guides                                                                       |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add . at the end of line.

File: applications/Parallel.md [hunk 16]
```diff
@@ -175,17 +175,16 @@ The major deliverable of for this milestone:
...
+|    6. | User Testing                        | We will conduct user testing to improve our product's UX and UI, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users use the v1 platform and provide a summary of the findings and improvements made based on insights                                                                                                                |
```
Comment (ayushmishra2005) [inline applications/Parallel.md]: Add . at the end of line.

File: applications/Parallel.md [hunk 17]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
--   **FTE:** 2
--   **Costs:** 5000 USD
+-   **FTE:** 1.8
+-   **Costs:** 4000 USD
 
 The major deliverable of for this milestone:
 
--   Develop cross-chain capabilities between DOT and KSM tokens
--   Users will be able to deposit DOT tokens as collateral and borrow KSM tokens through our web app
+-   Implement a cross-asset lending protocol.
```
Comment (kaichaosun) [inline applications/Parallel.md]: cross-asset -> multi-assets

File: applications/Parallel.md [hunk 18]
```diff
@@ -129,38 +129,38 @@ The lending protocol was inspired by compound protocol and our blockchain soluti
...
-|    3.c | Docker                     | We will provide a dockerfile to demonstrate the full cross-chain functionality                                                                                                                                                |
-|     4. | User Testing               | We will conduct user testing to improve our product's UX and UI, to ensure that the borrowing and lending functionalities are intuitive. Initially, we will conduct qualitative user testing by observing 10-15 users use the v1 platform and provide a summary of the findings and improvements made based on insights.                                                                            |
+|    1.a | Substrate module: Loans pallet | Loans pallet will be implemented as a cross-asset lending protocol which offers lending and borrowing by using floating rate model.                                                                                     |
```
Comment (kaichaosun) [inline applications/Parallel.md]: multi-assets

**Review by kaichaosun — APPROVED** [review]

**Comment by yz89** [comment]
Thanks for the comments, and I have fixed all the issues. @ayushmishra2005 @kaichaosun

File: applications/Parallel.md (hunk 6 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: This is not fixed.

File: applications/Parallel.md (hunk 7 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: Please fix this

File: applications/Parallel.md (hunk 8 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: Please address this comment

File: applications/Parallel.md (hunk 11 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: Please address this comment

File: applications/Parallel.md (hunk 9 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: Please address this comment

**Comment by yz89** [comment]
What do you think about the explanation we give? @kaichaosun @ayushmishra2005

**Comment by ayushmishra2005** [comment]
@yz89 Some minor comments are still left. Please address them.

File: applications/Parallel.md (hunk 11 above)
Comment (ayushmishra2005) [reply applications/Parallel.md]: @yz89 Can you please address this as well?

**Review by ayushmishra2005 — APPROVED** [review]

**Pull request merged**